numpy/__init__.py

```diff
@@ -232,9 +241,20 @@
     __all__.extend(['__version__', 'show_config'])
     __all__.extend(core.__all__)
     __all__.extend(_mat.__all__)
-    __all__.extend(lib.__all__)
     __all__.extend(['linalg', 'fft', 'random', 'ctypeslib', 'ma'])
 
+    _lib_expired_utils_names = ["byte_bounds", "safe_eval", "who"]
+    lib__all__ = lib.__all__[:]
+    for name in _lib_expired_utils_names:
+        lib__all__.remove(name)
+        __expired_functions__[name] = (
+            f"{name} is deprecated and removed from the "
+            f"main namespace. If you still want to use it "
+            f"please use it directly from np.lib or np.lib.utils"
```

Comment:
You want to leave them in the main namespace for now, to avoid breaking backwards compatibility right away.

Reply:
I added it back - the warning is raised only on calling them.